packages/flutter_tools/lib/src/ios/migrations/ios_migrator.dart

```diff
@@ -6,20 +6,22 @@ import 'package:meta/meta.dart';
 
 import '../../base/file_system.dart';
 import '../../base/logger.dart';
+import '../../reporting/reporting.dart';
 
 /// iOS project is generated from a template on Flutter project creation.
 /// Sometimes (due to behavior changes in Xcode, CocoaPods, etc) these files need to be altered
 /// from the original template.
-class IOSMigrator {
-  IOSMigrator(this.logger);
+abstract class IOSMigrator {
```

Comment:
Why does `processFileLines` need to take a callback? Could it call an abstract method provided by the subclasses?

Reply:
The migration can handle multiple files with different pattern matching per file:

```dart
  void run() {
    _migrateXcodeProjectInfoFile();
    _migratePodfile();
  }
 void _migrateXcodeProjectInfoFile() {
    _processFileLines(_xcodeProjectInfoFile, (String line) {
      if (line.contains('3B80C3941E831B6300D905FE')) {
        return null;
      }
...
    });
  }

  void _migratePodfile() {
    _processFileLines(_podfile, (String line) {
      if (line.contains('Prevent Cocoapods from embedding a second Flutter framework and causing an error with the new Xcode build system.')) {
        return null;
      }
...
    });
  }
```
Original version of `RemoveFrameworkLinkAndEmbeddingMigration` (Podfile migration since deleted):
https://github.com/jmagman/flutter/blob/e70ed529c41b1dbfe6392c45f84b6cfb165fb419/packages/flutter_tools/lib/src/ios/ios_project_migration.dart